### ISSUE TITLE
Error: Failed to understand configuration file: Unrecognized field "roles"

### ISSUE THREAD

**Issue by l0u1sg** [EpiLink/EpiLink#331]
Hello (it's me again :))

I'm testing the arm version of Epilink and I have a problem with the configuration, I can't get the roles option into the Discord object. Here is the error code from my configuration:

```
2022-01-22 09:11:47.284 ERROR epilink.main - Failed to understand configuration file: Unrecognized field "roles" (class org.epilink.bot.config.DiscordConfiguration), not marked as ignorable (6 known properties: "servers", "stickyRoles", "defaultLanguage", "commandsPrefix", "welcomeUrl", "preferredLanguages"])
 at [Source: (BufferedReader); line: 63, column: 1] (through reference chain: org.epilink.bot.config.Configuration["discord"]->org.epilink.bot.config.DiscordConfiguration["roles"])
```
```
rulebook: |
  "EmailIsInDomain" % { email ->
    // This code gets executed whenever the EmailIsInDomain rule is applied
    if (email.endsWith("@tumo.world")) {
        roles += "TUMOStudent"
    }
    if (email.endsWith("@tumo.center")) {
        roles += "TUMOTeach"
    }
[...]
discord:
  welcomeUrl: "https://tumoauth..."
  commandsPrefix: "e?"
  defaultLanguage: fr
  preferredLanguages: ["fr"]
  roles:
    - name: TUMOStudent
      rule: EmailIsInDomain
    - name: TUMOTeach
      rule: EmailIsInDomain
  servers:
    - id: 7797574...
      roles:
        TUMOStudent: 779765...
        TUMOTeach: 77976489...
  stickyRoles: [...] # optional
```

Is it me doing the configuration wrong or a bug on the beta?

**Comment by utybo** [comment]
Hello again! The beta changed the format for applying rules. (This is a breaking change from 0.6 that is described in the changelog)

This isn't described in the README yet (will be with the AMR PR) but the docs site was significantly improved for 0.7, you can see the current beta version here: https://epilink-beta.zoroark.guru

TL;DR, in your case, use `requires: [EmailIsInDomain]` instead of the `roles` block.

**Comment by l0u1sg** [comment]
Right, it works.
We agree that with the current configuration, if the person who connects with Google to a mail in @tumo.world will have a role, if it's a mail in @tumo.center, he will have another role, and if it's another mail, he will have no role added?

**Comment by l0u1sg** [comment]
I've managed to configure everything and it works great! Thanks again for your help and all the best for Epita 🚀

**Comment by utybo** [comment]
That's great to hear! Feel free to ask if you have further questions :)

Apologies, was unavailable to answer your second-to-last comment. FYI we have a "rule tester" you can use to try out your rules (although I now realize it may not work for rulebooks defined *in* the configuration and only for rulebooks defined in a separate .rule.kts file), you can check it out [here](https://epilink-beta.zoroark.guru/#/Admin/IRT).

You can run it via Docker with:

```sh
$ docker run -ti --rm -v "/path/to/folder/containing/rulebook:/var/run/eipilink/config/rules" docker.io/<image_here> bin/epilink-backend -t config/rules/your_rulebook.rule.kts
```
